pkg/metadata/service_metadata.go

```diff
@@ -0,0 +1,118 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package metadata
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	"github.com/ghodss/yaml"
+)
+
+// ServiceMetadata consists of information about the service and relative links as well
+// as a list of supported/deprecated versions
+type ServiceMetadata struct {
+	Service ServiceDetails `json:"service"`
+	// CRDs to ignore. ACK generator would skip these resources.
+	Versions []ServiceVersion `json:"versions"`
+}
+
+// ServiceDetails contains string identifiers and relevant links for the
+// service
+type ServiceDetails struct {
+	// The full display name for the service. eg. Amazon Elastic Kubernetes
+	// Service
+	FullName string `json:"full_name"`
+	// The short name (abbreviation) for the service. eg. S3
+	ShortName string `json:"short_name"`
+	// The URL of the service's homepage
+	Link string `json:"link"`
+	// The URL of the service's main documentation/user guide
+	Documentation string `json:"documentation"`
+}
+
+// ServiceVersion describes the status of all existing version of the controller
+type ServiceVersion struct {
+	APIVersion string    `json:"api_version"`
+	Status     APIStatus `json:"status"`
+}
+
+func (m *ServiceMetadata) GetLatestAPIVersion() (string, error) {
+	availableVersions := m.GetAvailableAPIVersions()
+
+	if len(availableVersions) == 0 {
+		return "", fmt.Errorf("service metadata contains no available versions")
```

Comment:
suggest moving the `fmt.Errorf` to being an `errors.New` and make it a package-level exported variable.

same for the other places you use `fmt.Errorf` in this file.